.github/workflows/rust-build-and-test.yml

```diff
@@ -17,9 +17,9 @@ env:
 jobs:
   check:
     runs-on: ubuntu-latest
+    container: 'public.ecr.aws/r5b3e0r5/3box/rust-builder:latest'
```

Comment:
This changes our actions to using our own builder image to simplify installing dependencies. I seem to remember we had that initially and had to change away from it. Does anyone remember why? It will be much simpler if we do not have to maintain two separate build images.

Reply:
I don't remember the reason we changed away either. Could it have been related to GH/cargo caching?

Reply:
I checked and GH/cargo caching is still working so I guess we will go forward and if it becomes a problem again we can fix it.